### PR TITLE
Merge various upstream fixes

### PR DESCRIPTION
**Pull request by schwartzdev**
cc: @kknd113 @quetzaluz

### CODE REVIEW

File: api/app/helpers/spree/api/api_helpers.rb [hunk 1]
```diff
@@ -83,7 +83,7 @@ def required_fields_for(model)
 
       @@payment_attributes = [
         :id, :source_type, :source_id, :amount, :display_amount,
-        :payment_method_id, :response_code, :state, :avs_response, :created_at,
```
Comment (quetzaluz) [inline api/app/helpers/spree/api/api_helpers.rb:86]: In checkouts_controller_decorator.rb we set the :response_code attribute (with the braintree payment token), so I believe we might need to remove this change here.

If this change is absolutely necessary, let me know and I will concat this attribute in the `Spree::Api::CheckoutsController#set_strong_parameters` method in dotandbo-spree.

**Comment by sankalpk** [comment]
In Flow I specify RABL templates using the template parameter. One of the commits might break that ability.

File: core/app/models/spree/product.rb [hunk 2]
```diff
@@ -97,6 +97,9 @@ class Product < Spree::Base
 
     alias :options :product_option_types
 
+    self.whitelisted_ransackable_associations = %w[stores variants_including_master master variants]
```
Comment (quetzaluz) [inline core/app/models/spree/product.rb:100]: We may need to add `sku` to this list due to the following call to ransack in dotandbo_spree:
`Spree::Product.ransack("variants_including_master_sku_cont" => keywords).result`

Comment (quetzaluz) [inline core/app/models/spree/product.rb:100]: > We may need to add sku to this list due to the following call to ransack in dotandbo_spree:
> Spree::Product.ransack("variants_including_master_sku_cont" => keywords).result

Oops, nevermind, misread that ransack statement, no need to change, it's covered here:
https://github.com/dotandbo/spree/pull/22/files#diff-f61dbf6358707ede7a8f24c85692baa5R54

**Comment by schwartzdev** [comment]
@sankalpk can you whitelist your use case?

**Comment by schwartzdev** [comment]
@quetzaluz I removed the payment_method change

**Comment by quetzaluz** [comment]
reviewed and :+1: from me

**Comment by kknd113** [comment]
reviewed

**Comment by sankalpk** [comment]
@schwartzdev, I whitelisted Flow related ransack attributes and also restricted the ability to specify templates to only admin API users.

**Comment by sankalpk** [comment]
If the changes above suffice, then :+1:  for me

**Comment by schwartzdev** [comment]
Looks good.

**Pull request merged**